lib/compress/zstd_double_fast.c

```diff
@@ -312,8 +359,12 @@ size_t ZSTD_compressBlock_doubleFast_dictMatchState_generic(
         U32 offset;
         size_t const h2 = ZSTD_hashPtr(ip, hBitsL, 8);
         size_t const h = ZSTD_hashPtr(ip, hBitsS, mls);
-        size_t const dictHL = ZSTD_hashPtr(ip, dictHBitsL, 8);
-        size_t const dictHS = ZSTD_hashPtr(ip, dictHBitsS, mls);
+        size_t const dictHashAndTagL = ZSTD_hashPtr(ip, dictHBitsL + ZSTD_SHORT_CACHE_TAG_BITS, 8);
+        size_t const dictHashAndTagS = ZSTD_hashPtr(ip, dictHBitsS + ZSTD_SHORT_CACHE_TAG_BITS, mls);
+        U32 const dictMatchIndexAndTagL = dictHashLong[dictHashAndTagL >> ZSTD_SHORT_CACHE_TAG_BITS];
+        U32 const dictMatchIndexAndTagS = dictHashSmall[dictHashAndTagS >> ZSTD_SHORT_CACHE_TAG_BITS];
+        size_t const dictTagsMatchL = (dictMatchIndexAndTagL & ZSTD_SHORT_CACHE_TAG_MASK) == (dictHashAndTagL & ZSTD_SHORT_CACHE_TAG_MASK);
+        size_t const dictTagsMatchS = (dictMatchIndexAndTagS & ZSTD_SHORT_CACHE_TAG_MASK) == (dictHashAndTagS & ZSTD_SHORT_CACHE_TAG_MASK);
```

Comment:
These are `bool`s, they don't need to be a `size_t`. Should probably just be `int`s?

Reply:
Yeah, an `int` would best convey the meaning of a bool.

Though this is entirely for readability, the compiler shouldn't care.